pkg/resource/fake/terraformed.go

```diff
@@ -40,6 +41,11 @@ func (o *Observable) SetObservation(data map[string]interface{}) error {
 	return nil
 }
 
+// GetCustomConnectionKeys is a mock
+func (o *Observable) GetCustomConnectionKeys(_ map[string]interface{}) (map[string][]byte, error) {
```

Comment:
Changed the name as you suggested but ended up using the same signature because I don't want to open a door where existing keys are overridden by custom keys defined by the provider developer. Right now, I check it and return an explicit error that this is not supported and this could be done in a more straightforward way with this signature.